### PR TITLE
refactoring SceneXplain plugin

### PR DESCRIPTION
**Pull request by delgermurun**
As mentioned [here](https://github.com/Significant-Gravitas/Auto-GPT-Plugins/pull/54#discussion_r1177429033), created a class.

Also improved description and name.

### CODE REVIEW

File: src/autogpt_plugins/scenex/__init__.py [hunk 1]
```diff
@@ -18,24 +21,25 @@ class AutoGPTSceneXPlugin(AutoGPTPluginTemplate):
...
         )
+        self.scenexplain = SceneXplain(os.getenv("SCENEX_API_KEY"))
+        self._is_api_key_set = bool(os.getenv("SCENEX_API_KEY"))
 
-        if is_api_key_set():
+    def post_prompt(self, prompt: PromptGenerator) -> PromptGenerator:
+        if self._is_api_key_set:
             prompt.add_command(
-                "Describe image by URL",
+                self._description,
```
Comment (ntindle) [inline src/autogpt_plugins/scenex/__init__.py:37]: This is a bit long for reuse potentially. Willing to give it a shot

File: src/autogpt_plugins/scenex/__init__.py [hunk 2]
```diff
@@ -18,24 +21,25 @@ class AutoGPTSceneXPlugin(AutoGPTPluginTemplate):
...
-        from .scenex_plugin.scenex_plugin import (
-            describe_image,
-            is_api_key_set,
+        self._description = (
+            "An Image Captioning Tool: Use this tool to generate a detailed caption for an image. "
+            "The input can be an image file of any format, and "
+            "the output will be a text description that covers every detail of the image."
         )
+        self.scenexplain = SceneXplain(os.getenv("SCENEX_API_KEY"))
+        self._is_api_key_set = bool(os.getenv("SCENEX_API_KEY"))
```
Comment (ntindle) [inline src/autogpt_plugins/scenex/__init__.py:32]: nit: you could optimize this to not check twice registering the value within the class

File: src/autogpt_plugins/scenex/__init__.py [hunk 3]
```diff
@@ -18,24 +21,25 @@ class AutoGPTSceneXPlugin(AutoGPTPluginTemplate):
...
 
-        if is_api_key_set():
+    def post_prompt(self, prompt: PromptGenerator) -> PromptGenerator:
+        if self._is_api_key_set:
```
Comment (ntindle) [inline src/autogpt_plugins/scenex/__init__.py:35]: re: setting the value in the class nit, `self.sceneexplain.is_api_key_set`

**Review by ntindle — COMMENTED** [review]
First pass looks good, will need to run tests later b/c at laptop not pc

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team to all open Plugin-PRs

We're in the process of re-architecting:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

Please be patient as we work to clean up the AutoGPT core.

If you already have an existing plug-in that’s affected, please message @ntindle on Discord so we can get your plugin spinning on the new 
architecture.

If you're working on a plugin, please bear in mind that one of our primary goals in re-architecting is to make life MUCH easier for you. 
We've taken aboard feedback from plugin developers. We're on it, so please give us some time!

<3 The AutoGPT Team

**Review by ntindle — CHANGES_REQUESTED** [review]
Please update PR

File: src/autogpt_plugins/scenex/__init__.py (hunk 1 above)
Comment (delgermurun) [reply src/autogpt_plugins/scenex/__init__.py:37]: I am not sure but I guess AutoGTP uses command description inside LLM's context, right?
Explaining more clearly is better for choosing right command, I think.

Btw, we use same description on the [langchain integration](https://github.com/hwchase17/langchain/blob/d3ec00b5664798be74daa9bdc28ae7c2c86ee22e/langchain/tools/scenexplain/tool.py#L24).

**Review by ntindle — APPROVED** [review]

**Pull request merged**